_posts/2018/2018-06-18-bridging-the-gap-between-literate-programming-and-teaching-modules.markdown

```diff
@@ -0,0 +1,94 @@
+making the programs more robust, more portable, and easier to maintain than
+programs are solely written in high-level languages.
+
+The notebooks or rather the Jupyter Ecosystem have found their place within research,
+education and data science, among many other areas, particularly promoting
+knowledge transfer. I first started using the notebooks when I realized that they
+allowed me not to only write code in a number of programming languages but also
+allowed me to build a compelling narrative in which analysis, documentation, and data are
+presented together transparently. Since then I have used the notebooks
+for dissemination, education, prototyping, have developed ad-hoc extensions/widgets and
+had helped others to use the notebooks better.
+
+However, the emergence of novel technologies also brings a new set of challenges
+to overcome during their implementation and usage. This case study presents
+some of the work the members of ODK have done in order to bring literate programming to
+the classrooms.
+
+## Scenario
+Most universities in the UK have an institutional learning platform commonly used
+by lecturers to share materials with students.
+Such a platform only allows for upload and rendering of documents in .pdf, .txt,
+and .docx formats (among others). As science becomes more computationally driven
+more there is a greater need to increase the data and computational literacy of
+the students.
+Multiple lecturers and professors at the University of Sheffield wanted to migrate
+from using paid-license software to open source software such as Python and at the same time, use the notebooks in their class.
+In addition to the notebooks lecturers need a permanent site to upload the materials,
+class notes, and other resources so that students can revise at any point and  
+download the materials.
+
+## Possible solutions
+1. Use of the institutional learning platform and separate distribution of the notebooks.
+2. Use of cloud solutions such as CoCalc to manage their course content.
+3. Leverage the use of the Jupyter notebooks along with binder, Microsoft Azure notebooks and GitHub pages.
+
+The main problem with solution 1. is the duplicate point of access for both students
+and the people running the class increasing the maintenance burden.
+We have used CoCalc in Sheffield for a number of classes (e.g. Biomedical sciences, scalable machine learning) with great success in the past. But in that case only those enrolled in the course can access the materials so unless the materials are in a service like GitHub or GitLab they cannot be publicly distributed.
+
+
+## The implementation
+Thanks to `nbformat` it is possible to convert the notebooks into other formats, such as .pdf, .tex, .py, .html, .md and even reveal.js slides. On the other hand, it is possible to now perform regression tests on the notebooks thanks to the ODK developed tool `nbval`, and finally, GitHub supports and serves static websites generated using Jekyll.
```

Comment:
link to nbval here

Reply:
Before digging into the implementation -- which is rather technical -- you may want to develop a user perspective:
feature set, layout of the course repo, links to some example repos and how they are rendered. Maybe a few words
one how to get started. All of the above could alternatively be externalized into the template documentation, with just a link to there. Possibly so as well for the implementation section.